helix-core/src/main/java/org/apache/helix/task/AssignableInstanceManager.java

```diff
@@ -178,6 +178,9 @@ public void buildAssignableInstances(ClusterConfig clusterConfig, TaskDataCache
     }
     LOG.info(
         "AssignableInstanceManager built AssignableInstances from scratch based on contexts in TaskDataCache due to Controller switch or ClusterConfig change.");
+    if (_assignableInstanceMap.isEmpty()) {
+      LOG.warn("No assignable instance!");
```

Comment:
May I suggest let's add more info here? So we could differentiate between the 3 logs here.

Reply:
Thanks for the review.
The above INFO log should provide enough context to differentiate 3 cases. I think it's fine to keep it this way. Let me know if you are fine with it. Thanks